packages/map-toolkit/src/deckgl/viewport-scale/utils.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { distance } from '@turf/distance';
+import { UNIT_MAP } from './constants';
+import type { GetViewportScaleArgs } from './types';
+
+const numberFormatter = Intl.NumberFormat('en-US');
+
+export function getViewportScale({
+  bounds,
+  unit,
+  formatter = numberFormatter,
+}: GetViewportScaleArgs) {
+  if (!bounds) {
+    return '-- x --' as const;
+  }
+
+  const [minX, minY, maxX, maxY] = bounds;
+
+  const width = formatter.format(
+    Math.round(
+      distance([minY, minX], [minY, maxX], {
+        units: UNIT_MAP[unit],
+      }),
+    ),
+  );
+
+  const height = formatter.format(
+    Math.round(
+      distance([minX, minY], [minX, maxY], {
+        units: UNIT_MAP[unit],
+      }),
+    ),
+  );
+
+  return `${width} x ${height} ${unit.toUpperCase()}` as const;
```

Comment:
I don't think we need the `as const` here since the general return value of the function is a dynamic string